### PR TITLE
Core Data: Clear auto-draft titles on save if not changed explicitly.

### LINKED ISSUE #17628: Auto Draft set as post title
**Describe the bug**

When I create a new post in Gutenberg and publish with some content but **no** title, it gets saved with the title set to “Auto Draft”.

I have to remove the title again and change the slug when I really want a post without a title.

**To reproduce**
1. Create new post
2. Write some content
3. Publish post
4. See "Auto Draft" title.
5. View post on frontend, notice "Auto Draft" title and "auto-draft" slug.

**Expected behavior**

The post should have no title and its ID as the slug.

**Screenshots**

Screenshot 2019-09-27 at 11 11 54

Video: https://cloudup.com/c5bALGfxT_8

**Additional context**
- Chrome 77 on macOS
- Current Gutenberg master
- WordPress 5.3-beta1-46334 running Twenty Nineteen theme
- PHP 7.3 and MySQL 5.6 on Apache

The same issue with "Auto Draft" was reported in #6614, but the bigger issue there was a nginx misconfiguration.

#8770 is kinda related too.

### PR DESCRIPTION
**Pull request by epiqueras**
Fixes #17628

## Description

This PR fixes an issue where saving an `auto-draft` without a title would persist its default title, "Auto Draft", in the new `draft` or `published` post. The latter two statuses don't have an incoming filter to clear the "Auto Draft" title so it would show in the editor and confuse users.

## How has this been tested?

It was verified that saving or publishing new posts without titles no longer sets the title to "Auto Draft", but keeps it empty instead.

## Types of Changes

*Bug Fix:* New posts saved or published without a title will no longer have their title set to "Auto Draft".

## Checklist:

- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by mcsf** [comment]
Worth expanding E2E tests?

**Comment by mcsf** [comment]
(thanks for the quick dive into this!)

**Comment by epiqueras** [comment]
Yeah, added one here: https://github.com/WordPress/gutenberg/pull/17633/commits/c30ed8027b696093e0969904141acd7f97ddc8a2

**Review by youknowriad — APPROVED** [review]

File: packages/core-data/src/actions.js [hunk 1]
```diff
@@ -325,15 +325,16 @@ export function* saveEntityRecord(
...
-				kind === 'postType' &&
-				persistedRecord && persistedRecord.status === 'auto-draft' &&
-				! data.status
-			) {
-				data = { ...data, status: 'draft' };
+			if ( kind === 'postType' && persistedRecord && persistedRecord.status === 'auto-draft' ) {
+				if ( ! data.status ) {
+					data = { ...data, status: 'draft' };
+				}
+				if ( ! data.title || data.title === 'Auto Draft' ) {
```
Comment (TimothyBJacobs) [inline packages/core-data/src/actions.js]: Should Auto Draft be put through a translation function? It is translated in `get_default_post_to_edit()`. Not 100% sure if that is the same place the post title is getting set in Gutenberg.

Comment (epiqueras) [reply packages/core-data/src/actions.js]: How does that relate to this change? We are clearing the title here.

Comment (swissspidy) [reply packages/core-data/src/actions.js]: Well it compares the title with an English string, which might not work on a localized environment

Comment (epiqueras) [reply packages/core-data/src/actions.js]: Got it.

So is "Auto Draft" translated in this context? Where would we check?

Comment (swissspidy) [reply packages/core-data/src/actions.js]: Yes, as that string is coming from `get_default_post_to_edit()`.

So you'd need to compare it with `__( 'Auto Draft' )`, not just `'Auto Draft'`

Comment (TimothyBJacobs) [reply packages/core-data/src/actions.js]: It looks like it is. I just tried @swissspidy's replication steps on trunk in a non-english locale, and the title was set to `Automatisch gespeicherter Entwurf` on publish.

Comment (epiqueras) [reply packages/core-data/src/actions.js]: I'm trying to replicate it, but `data.title` is always `undefined` here unless you explicitly typed in the auto draft title, so the second check doesn't even run.

**Pull request merged**